### PR TITLE
use Sphinx-Tags and replace Taxonomy page

### LINKED ISSUE #49: rethink tags
documents have tags hardcoded in content with HTML to color them red, but no crossref links

EG: https://github.com/numfocus/DISCOVER-handbook/blob/jupyter-init/DISCOVER/06_childcare.md

```markdown
---
layout: page
title: Childcare
---

# Childcare

**Tags:** <span style="color:red">Parents and Carers</span>, <span style="color:red">Participant Diversity</span>, <span style="color:red">Scholarships</span>

✅ Offer financial support for childcare.
 - Build it into your budget. 
   - Link it to sponsorships, or university partners, or ...
```

There's a section in jupyter book docs to create cross references here:
* https://jupyterbook.org/en/stable/content/references.html
* https://jupyterbook.org/en/stable/tutorials/references.html

Need to do more research to see if this is useful for tags. Turns out jbook is built on top of Sphinx, so there are similarities.

here is a tool to create tags in jupyterbooks/sphinx: https://github.com/melissawm/sphinx-tags

### PR DESCRIPTION
**Pull request by mikofski**
- closes #49 
- installs sphinx-tags from @melissawm 
- replaces subsection tags with one `{tags}` role at the top of the page
- replaces the Taxonomy page with Tags Overview tree/pages

### CODE REVIEW

**Comment by mikofski** [comment]
A coupled of notes/decisions:
- [ ] okay to use all caps? and replace whitesapce with dashes?
- [ ] better at top of page or move to bottom?

**Review by terezaif — APPROVED** [review]
LGTM!

**Comment by mikofski** [comment]
Okay, I'll try to wrap this up asap, maybe in the next 2 weeks, but def before next meeting.

**Comment by mikofski** [comment]
This is done, I guess, but now the "glossary" page needs to be updated to actually be a glossary, see #61.

Also if you diff the actual tags and what's in the taxonomy there are some items that don't overlap, namely:

* Conference Logistics
  * Accommodation
  * Organization and Program Committees
  * Talks
  * Poster Sessions
  * CFP

* Diverse Groups
  * Racial Minorities

* Potential challenges
  * Discrimination

so don't know what to do with that?

**Comment by melissawm** [comment]
I know this is not necessarily in scope for this PR but I have a suggestion for the effort/impact quadrant in the front page:

````

```
                              |
                              |
             🍎               |               ✅
                              |
        High Impact,          |          High Impact,
                              |
        Low Effort            |          High Effort
                              |
                              |
— — — — — — — — — — — — — — — — — — — — — — — — — — — — — — — 
                              |
                              |
                              |
         Low Impact,          |           Low Impact,
                              |
         Low Effort           |           High Effort
                              |
                              |
```
````

This gives it a little more aligned look:

Screenshot_20230210_103057

File: DISCOVER/_config.yml [hunk 1]
```diff
@@ -21,3 +21,11 @@ repository:
...
   use_issues_button: true
   use_repository_button: true
+
+# config sphinx tags extension
+sphinx:
+  extra_extensions:
+   - sphinx_tags
+  config:
+    tags_create_tags: True
+    tags_extension: ["md"]  # using MyST, not default: ["rst"]
```
Comment (melissawm) [inline DISCOVER/_config.yml:31]:
```suggestion
    tags_extension: ["md"]  # using MyST, not default: ["rst"]
    suppress_warnings: ["etoc.toctree"]
```

This prevents the `WARNING: toctree directive not expected with external-toc [etoc.toctree]` (see melissawm/sphinx-tags#48)

**Review by melissawm — APPROVED** [review]
Looks good to me - the only caveat is that if you try to build the jupyter-book twice without deleting the _tags folder first, you may end up with duplicate items in the local toctrees of each individual tag page. I have opened an issue at sphinx-tags for this (melissawm/sphinx-tags#48) but this should not impact the deployment of the cookbook.

**Comment by mikofski** [comment]
@noatamir  if you like the look of the new tags, do you want to push the merge pr button? This will make it go live so hesitant to merge without you approval. thx

**Comment by noatamir** [comment]
@mikofski thanks for the tag. I was OOO all week and will look this over soon!
But I trust Tereza and Melissa if you don't want to wait =)

**Review by noatamir — APPROVED** [review]
LGTM. Maybe remove the glossary/taxonomy page. Can also be a separate PR.

**Comment by mikofski** [comment]
> Maybe remove the glossary/taxonomy page. Can also be a separate PR.

see #64 "remove glossary"

**Pull request merged**